### ISSUE TITLE
Handle base class member access from derived class in include-cleaner

### ISSUE THREAD

**Issue by hokein** [llvm/llvm-project#59251]
Given an example

```hpp
// base.h
class Base { int a; };
```

```hpp
// derived.h
class Derived : public Base {};
```

```cpp
// main.cpp
#include "derived.h"

void fun(Derived d) {
  d.a;
}
```
Should we include the `base.h` for the base member access `a`? The conclusion is no, as this member expr is via the Derived object.

In current include-cleaner library implementation, we consider the `base.h` is used (thus we should insert it) -- when we visit the `MemberExpr .a` in `WalkAST.cpp`, the corresponding target symbol is the `FieldDecl a` defined in Base which is in base.h.

For this case, we should not report the `FieldDecl a`, instead we should visit the type of the Base expr `d` which is `class Derived`. 

A sad bit is that the `RefLocation` and `Target` is confusing in the HTML dump tool now:
  - `RefLocation` points to `d.^a`;
  - `Target` points to the `Derived` class;

But this is not a thing we need to worry about at the moment, we can improve later on.
